### PR TITLE
simplify work report code logic

### PR DESCRIPTION
**Pull request by zikunfan**

### CODE REVIEW

File: cstrml/tee/src/lib.rs [hunk 1]
```diff
@@ -185,46 +186,42 @@ impl<T: Trait> Module<T> {
...
     pub fn update_identities() {
-        // Ideally, reported_rs should be current_rs + 1
-        let reported_rs = Self::get_reported_slot();
+        // 1. Update id's work rzeport, get id's workload and total workload
+        let mut total_meaningful_workload = 0;
+        let mut total_empty_workload = 0;
         let current_rs = Self::current_report_slot();
 
-        // 1. Report slot did not change, it should not trigger updating
-        if current_rs == reported_rs {
```
Comment (MyronFanQiu) [inline cstrml/tee/src/lib.rs:193]: I thint it make senses to have this check.

Comment (MyronFanQiu) [reply cstrml/tee/src/lib.rs:193]: anyway. It's ok to do it again.

File: cstrml/tee/src/lib.rs [hunk 2]
```diff
@@ -257,42 +255,18 @@ impl<T: Trait> Module<T> {
...
+        if let Some(wr) = Self::work_reports(controller) {
+            if Self::reported_in_slot(controller, current_rs) {
+                (wr.empty_workload as u128, wr.meaningful_workload as u128)
             } else {
-                // Cut workload
-                MeaningfulWorkload::mutate(|mw| *mw -= wr.meaningful_workload as u128);
-                EmptyWorkload::mutate(|ew| *ew -= wr.empty_workload as u128);
-                0
+                // Remove work report
+                <WorkReports<T>>::remove(controller);
```
Comment (MyronFanQiu) [inline cstrml/tee/src/lib.rs:265]: 300-600 no wr.
650. new wr
660 end of era. new wr would be deleted.
920 end of new era. Cannot find wr.
950 new wr.
1220 end of new era. wr is found and valid. work come back. 

I recommend checking the wr.block_number to see whether it's a new work report or not before deleting it.

Comment (zikunfan) [reply cstrml/tee/src/lib.rs]: that make sense, this will make new reporter confused.

**Pull request merged**